plugindocs.rb

```diff
@@ -17,6 +17,8 @@ class PluginDocs < Clamp::Command
 
   parameter "PLUGINS_JSON", "The path to the file containing plugin versions json"
 
+  ALIAS_MAPPINGS_URL = 'https://raw.githubusercontent.com/elastic/logstash/master/logstash-core/src/main/resources/org/logstash/plugins/AliasRegistry.yml'
```

Comment:
Right now, before the merge of this PR, `AliasRegistry` is an internal detail of Logstash core, and merging this PR would effectively make this doc tool dependent upon the current shape. 

Would it be possible to open the separate PR against logstash core's `AliasRegistry` as we discussed, to refactor it to include the replacements (and to still generate the exisiting fingerprinted output expected by the plugin manager)?

If we can get that merged before this PR, we won't need to do a compatibility dance where docs are broken until everything is merged.

From the structure we need, I would expect something like:

~~~ yaml
input: # key indicates plugin type, must be one of `input`, `output`, `codec`, or `filter`.
  elastic_agent: # key indicates alias' name
    from: beats # key `from` is _required_, string value indicates source plugin name
    docs: # docs key is _optional_, contains an array of docs-oriented instructions
    - replace: ":plugin: beats" # as of now, a docs instruction must be a key/value map with `replace` and `with` keys, each containing a string. We can extend this in the future if we need to.
      with: ":plugin: elastic_agent"
    - replace: ":plugin-uc: Beats"
      with: ":plugin-uc: Elastic Agent"
    - replace: ":plugin-singular: Beat"
      with: ":plugin-singular: Elastic Agent"
~~~

Reply:
Thanks for the review @yaauie , appreciate!

`AliasRegistry` update in logstash core is almost ready, working on unit tests now and thinking to update & merge this PR after that.
[Here is a gist](https://gist.github.com/mashhurs/b073ab8d530fc2d4d8de6e6c69eec960) if you want to take a quick glance.

Reply:
Here is the PR: [link](https://github.com/elastic/logstash/pull/13971)

Reply:
Alias definition PR was merged and I applied the required changes here. Now, no any hard coded header declarations, all come through the YAML alias definition.